generators/base-application/tasks.d.mts

```diff
@@ -70,43 +55,45 @@ type ClientSource<ExtendsSelf extends ClientSource = ClientSource> = {
-    // Add application to existing priorities
-    Record<
-      | 'loadingTaskParam'
-      | 'preparingTaskParam'
-      | 'defaultTaskParam'
-      | 'writingTaskParam'
-      | 'postWritingTaskParam'
-      | 'preConflictsTaskParam'
-      | 'installTaskParam'
-      | 'postInstallTaskParam'
-      | 'endTaskParam',
-      ApplicationTaskParam<Definition>
-    > &
-    // Add entities to existing priorities
-    Record<'defaultTaskParam', EntitiesTaskParam<Definition>> &
-    // Add application and control to new priorities
-    Record<
-      | 'configuringEachEntityTaskParam'
-      | 'loadingEntitiesTaskParam'
-      | 'preparingEachEntityTaskParam'
-      | 'preparingEachEntityFieldTaskParam'
-      | 'preparingEachEntityRelationshipTaskParam'
-      | 'postPreparingEachEntityTaskParam'
-      | 'writingEntitiesTaskParam'
-      | 'postWritingEntitiesTaskParam',
-      ControlTaskParam & ApplicationTaskParam<Definition>
-    > & {
-      // Add additional types to each priority
-      applicationType: Definition['applicationType'];
-      configuringEachEntityTaskParam: ConfiguringEachEntityTaskParam;
-      loadingEntitiesTaskParam: LoadingEntitiesTaskParam;
-      preparingEachEntityTaskParam: EachEntityTaskParam<Definition>;
-      preparingEachEntityFieldTaskParam: PreparingEachEntityFieldTaskParam<Definition>;
-      preparingEachEntityRelationshipTaskParam: PreparingEachEntityRelationshipTaskParam<Definition>;
-      postPreparingEachEntityTaskParam: EachEntityTaskParam<Definition>;
-      writingEntitiesTaskParam: EntitiesTaskParam<Definition>;
-      postWritingEntitiesTaskParam: SourceTaskParam<Definition> & EntitiesTaskParam<Definition>;
-    };
+export type BaseApplicationGeneratorDefinition<
+  Definition extends { applicationType: any; entityType: any; sourceType: any } = GenericApplicationDefinition &
+    GenericSourceTypeDefinition<Record<string, (...args: any[]) => void>>
+> = BaseGeneratorDefinition<Definition> &
+  // Add application to existing priorities
+  Record<
+    | 'loadingTaskParam'
+    | 'preparingTaskParam'
+    | 'defaultTaskParam'
+    | 'writingTaskParam'
+    | 'postWritingTaskParam'
+    | 'preConflictsTaskParam'
+    | 'installTaskParam'
+    | 'postInstallTaskParam'
+    | 'endTaskParam',
+    ApplicationTaskParam<Definition>
+  > &
+  // Add entities to existing priorities
+  Record<'defaultTaskParam', EntitiesTaskParam<Definition>> &
+  // Add application and control to new priorities
+  Record<
+    | 'configuringEachEntityTaskParam'
+    | 'loadingEntitiesTaskParam'
+    | 'preparingEachEntityTaskParam'
+    | 'preparingEachEntityFieldTaskParam'
+    | 'preparingEachEntityRelationshipTaskParam'
+    | 'postPreparingEachEntityTaskParam'
+    | 'writingEntitiesTaskParam'
+    | 'postWritingEntitiesTaskParam',
+    ControlTaskParam & ApplicationTaskParam<Definition>
+  > & {
+    // Add additional types to each priority
+    applicationType: Definition['applicationType'];
```

Comment:
I may have misundertood the feature: Isn't it confusing to mix tasks param definition and type?

Reply:
It should contain every generator customizable type, not only tasks.
Not surer it's used, but it's there for convenience.